src/Workspaces/Core/Portable/Remote/RemoteArguments.cs

```diff
@@ -52,20 +52,25 @@ public bool Equals(SerializableSymbolAndProjectId other)
             => other != null && SymbolKeyData.Equals(other.SymbolKeyData) && ProjectId.Equals(other.ProjectId);
 
         public static SerializableSymbolAndProjectId Dehydrate(
-            IAliasSymbol alias, Document document)
+            IAliasSymbol alias, Document document, CancellationToken cancellationToken)
         {
             return alias == null
                 ? null
-                : Dehydrate(new SymbolAndProjectId(alias, document.Project.Id));
+                : Dehydrate(document.Project.Solution, alias, cancellationToken);
         }
 
         public static SerializableSymbolAndProjectId Dehydrate(
-            SymbolAndProjectId symbolAndProjectId)
+            Solution solution, ISymbol symbol, CancellationToken cancellationToken)
         {
+            var symbolKey = symbol.GetSymbolKey(cancellationToken);
+            var projectId = solution.GetExactProjectId(symbol);
+            if (projectId == null)
+                throw new InvalidOperationException("SymbolKeys used for OOP operations must have a ProjectId");
+
             return new SerializableSymbolAndProjectId
             {
-                SymbolKeyData = symbolAndProjectId.Symbol.GetSymbolKey().ToString(),
-                ProjectId = symbolAndProjectId.ProjectId
+                SymbolKeyData = symbolKey.ToString(),
+                ProjectId = projectId,
             };
```

Comment:
this type (SerializableSymbolAndProjectId) still stays around.  It's the marshaling type we use to be able to talk about a symbol and ensure it can be rehydrated on hte other side.  However, SymbolAndProjectId will go away.